### ISSUE TITLE
Release builds should be notarized for macOS Catalina

### ISSUE THREAD

**Issue by shs96c** [bazelbuild/bazel#10777]
### Description of the problem / feature request:

In macOS Catalina, Apple introduced a requirement that applications that are downloaded and run on the machine are notarized. Without this, applications refuse to run unless a user follows a non-obvious sequence of steps involving the "Security" preference pane.

Bazel releases should be notarized in order to ensure that it works as expected on macOS. 

### What operating system are you running Bazel on?

macOS Catalina and above

###  Have you found anything relevant by searching the web?

Mozilla also ran into (and resolved!) this problem with both Firefox and `geckodriver`:

* https://firefox-source-docs.mozilla.org/testing/geckodriver/Notarization.html
* https://bugzilla.mozilla.org/show_bug.cgi?id=1588081

Steps for notarizing applications on the command line can be found here: https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution/customizing_the_notarization_workflow

There may already be machinery within Google that can be harnessed for notarizing releases, which may simplify this issue considerably.

**Comment by davido** [comment]
Duplicate: #9304.

**Comment by aiuto** [comment]
The title here says it better than #9304 
I gave this to both the Apple and EngProd teams since it may require collaboration to get the nuances.

**Comment by tetromino** [comment]
Yep, this affects the latest MacOS Bazel installer if it's downloaded using the browser (as opposed to curl or wget). We should fix this, and until we fix this, we should document a workaround.

**Comment by tetromino** [comment]
Correction: this affects only the Bazel _executable_ (e.g. `bazel-2.1.0-darwin-x86_64`).

The _installer script_ (which is what we suggest users to download in https://docs.bazel.build/versions/master/install-os-x.html) works fine when downloaded from the browser.

Downgrading priority to P2.

**Comment by meteorcloudy** [comment]
I'm moving our release pipeline to Kokoro so that we can do the signing and notarization for macOS.

**Comment by fweikert** [comment]
@meteorcloudy what's the current state of the Kokoro builds?

**Comment by meteorcloudy** [comment]
They are set up, but we haven't switched to them because we still haven't figured out the notarization process..

**Comment by jin** [comment]
Is this still a P1 bug?

**Comment by meteorcloudy** [comment]
It doesn't look like so. Deprioritizing this to P2.
